synapseclient/table.py

```diff
@@ -607,6 +609,37 @@ def join_column_names(columns):
     return ",".join(escape_column_name(c) for c in columns)
 
 
+def _convert_df_date_cols_to_datetime(
+    df: DataFrameType, date_columns: List
+) -> DataFrameType:
+    """Convert date columns with epoch time to date time in UTC timezone
+    :param df: a pandas dataframe
+    :param date_columns: name of date columns
+    """
+    test_import_pandas()
+    import pandas as pd
+    import numpy as np
+
+    # find columns that are in date_columns list but not in dataframe
+    diff_cols = list(set(date_columns) - set(df.columns))
+    if diff_cols:
+        raise ValueError("Please ensure that date columns are already in the dataframe")
+    try:
+        df[date_columns] = (
+            df[date_columns].fillna(-1).astype(np.float64).astype(np.int64)
```

Comment:
I like the idea, but unfortunately -1 is a valid epochtime.  On the off chance someone actually puts in -1 as a value, you would replace it to a blank in line 636.  What are some other ways you would support NA values?